src/stream/src/executor/backfill/no_shuffle_backfill.rs

```diff
@@ -295,6 +302,10 @@ where
                                         break 'backfill_loop;
                                     }
                                     Some(chunk) => {
+                                        tokio::time::sleep(Duration::from_millis(
+                                            self.snapshot_read_delay as u64,
+                                        ))
+                                        .await;
```

Comment:
Fyi @chenzl25 with this change, we can trigger delay in snapshot read arm during our tests.

Useful in the case of testing cancellation of stream jobs (as in this PR), and generally when our cluster is in-memory and we want to test backfill. This is because it can simulate some I/O delay with s3.